### ISSUE TITLE
Cargo included in Rust 1.1 can no longer build in Windows NTFS junctions

### ISSUE THREAD

**Issue by Zarathustra30** [rust-lang/cargo#1749]
The following script creates a new project, compiles/runs it, creates a NTFS junction to the project, and attempts to compile/run the project through the NTFS junction. It runs successfully with the version included in Rust 1.0, but fails with the version included in Rust 1.1, with the error `Could not find Cargo.toml in 'path\to\example\linked_project'`.
```
@echo off
cargo new normal_project --bin
cd normal_project
cargo run
cd ..
mklink /J linked_project normal_project
cd linked_project
dir
cargo run
cd ..
```

(This script will fail if there already exists a `normal_project` or `linked_project` and fails to clean after itself)

As a side note, Cargo probably shouldn't have to compile this twice, but that may be expected behavior.

**Comment by alexcrichton** [comment]
This appears to be fixed for me with the nightly cargo, could you confirm that?

**Comment by Zarathustra30** [comment]
It is *sort of* fixed in 1.2 beta and 1.3 nightly. The script passes. However, if `normal_project` has never been built in the first place, cargo fails with an unknown error. `--verbose` tells me it is `os error 183` "Cannot create file when that file already exists.

New test batch:
```
@cargo new normal_project --bin
@mklink /J linked_project normal_project
@cd linked_project
@dir
cargo run ::--verbose
@cd ..
@rmdir linked_project
```
* Make sure there is *no* `normal_project`. It passes if it has been built outside of the junction.

* To enable verbose, delete the `::` in front of it.

* Passes on Rust 1.0 (efb482d 2015-04-30) (compiles and prints "Hello, world!")

* Fails on Rust 1.1 (a483581 2015-05-14) ("Could not find Cargo.toml in  'C:\path\to\linked_project'")

* Fails on Rust 1.2beta+ (a2d2d2ac 2015-06-24) ("An unknown error occurred") (--verbose says it is due to "os error 183")

*

**Comment by alexcrichton** [comment]
Ok I think I've found the underlying bug, which I believe is https://github.com/rust-lang/rust/issues/26716, so I'm going to close this in favor of that. Thanks for the report though!
